state/src/interface/mod.rs

```diff
@@ -147,63 +84,60 @@ impl State {
         tx.setup_tables()?;
         tx.commit()?;
 
-        // Create a proposal queue and the raft config
-        let (proposal_send, proposal_receiver) = unbounded();
-        let (raft_cancel_sender, raft_cancel_receiver) = new_cancel_channel();
-        let replication_config = ReplicationNodeConfig {
-            tick_period_ms: DEFAULT_TICK_INTERVAL_MS,
-            relayer_config: config.clone(),
-            raft_config,
-            proposal_receiver,
-            network,
+        // Setup the state machine
+        let applicator_config = StateApplicatorConfig {
+            allow_local: config.allow_local,
+            cluster_id: config.cluster_id.clone(),
             task_queue,
             handshake_manager_queue,
             db: db.clone(),
             system_bus: system_bus.clone(),
-            cancel_channel: raft_cancel_receiver,
         };
+        let applicator = StateApplicator::new(applicator_config).map_err(StateError::Applicator)?;
+        let notifications = OpenNotifications::new();
+        let sm_config = StateMachineConfig::new(config.raft_snapshot_path.clone());
+        let sm = StateMachine::new(sm_config, notifications.clone(), applicator);
 
-        // Start the Raft worker
-        let mut raft_worker = ReplicationNodeWorker::new(replication_config)
-            .expect("failed to build Raft replication node");
-        raft_worker.start().expect("failed to start Raft replication node");
+        // Start a raft
+        let raft = RaftClient::new(raft_config, db.clone(), network, sm)
```

Comment:
If the `openraft` thread ever fails/panics, how do we propagate this to the node?

Reply:
Not much we can do, caller would panic given that it's async. ITT we should observe it and see if that's a problem, panics in libraries should be rare and I'm wary of going out of our way to detect panics unless we believe it's a problem